tools/deploy/manager.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package deploy
+
+import "github.com/m3db/m3cluster/services"
+
+// DeploymentInstance represents an instance in the deployment.
+type DeploymentInstance interface {
+	// ID returns the instance id.
+	ID() string
+
+	// Revision returns the revision deployed to the instance.
+	Revision() string
```

Comment:
That's probably insufficient if we want to make the revision type truly generic, I'd actually make it an interface with `Validate` and `Equals` so users can provide their own implementation. Otherwise we'd end up doing string comparisons in the code anyway.

I think using string for revision is probably okay, and if the need for a more generic revision type arises in the future, can always easily change it.